bundles/core/org.openhab.core.library/src/main/java/org/openhab/core/library/types/PointType.java

```diff
@@ -0,0 +1,158 @@
+
+
+import org.apache.commons.lang.StringUtils;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.ComplexType;
+import org.openhab.core.types.PrimitiveType;
+import org.openhab.core.types.State;
+
+
+/**
+ * This type can be used for items that are dealing with GPS functionality.
+ * 
+ * @author Gaël L'hopital
+ * @since 1.7.0
+ */
+public class PointType implements ComplexType, Command, State {
+	
+	public static final double EARTH_GRAVITATIONAL_CONSTANT = 3.986004418e14;
+	public static final double WGS84_a = 6378137;				// The equatorial radius of WGS84 ellipsoid (6378137 m).
+	
+	private double latitude;									// in decimal degrees
+	private double longitude;									// in decimal degrees 
+	private double altitude = 0;								// in decimal meters
+	
+	// constants for the constituents
+	static final public String KEY_LATITUDE = "lat";
+	static final public String KEY_LONGITUDE = "long";
+	static final public String KEY_ALTITUDE = "alt";
+		
+	public static final State EMPTY = new PointType(new DecimalType(0), new DecimalType(0));
+	
+	public PointType(DecimalType latitude, DecimalType longitude) {
+		this.latitude = latitude.doubleValue();
+		this.longitude = longitude.doubleValue();
+		canonicalize();
+	}
+	
+	public PointType(DecimalType latitude, DecimalType longitude, DecimalType altitude) {
+		this(latitude, longitude);
+		this.altitude = altitude.doubleValue();
+	}
+	
+	public PointType(StringType latitude, StringType longitude) {
+		this(new DecimalType(latitude.toString()), new DecimalType(longitude.toString()));
+	}
+	
+	public PointType(StringType latitude, StringType longitude, StringType altitude) {
+		this(new DecimalType(latitude.toString()), new DecimalType(longitude.toString()), new DecimalType(altitude.toString()));
+	}
+	
+	public PointType(String value) {
+		if (StringUtils.isNotBlank(value)) {
+			String[] elements = value.split(",");
+			if (elements.length >= 2) {
+				this.latitude = Double.parseDouble(elements[0]);
+				this.longitude = Double.parseDouble(elements[1]);
+				if (elements.length == 3) {
+					this.altitude = Double.parseDouble(elements[2]);
+				}
+			}
+		}
+	}
+		
+	public DecimalType getLatitude() {
+		return new DecimalType(latitude);
+	}
+	
+	public DecimalType getLongitude() {
+		return new DecimalType(longitude);
+	}
+	
+	public DecimalType getAltitude() {
+		return new DecimalType(altitude);
+	}
+	
+	public void setAltitude(DecimalType altitude) {
+		this.altitude = altitude.doubleValue();
+	}
+	
+	public DecimalType getGravity() {
+		double latRad = Math.toRadians(latitude);
+	    double deltaG = -2000.0 * (altitude/1000) * EARTH_GRAVITATIONAL_CONSTANT / ( Math.pow(WGS84_a,3.0) );
+        double sin2lat = Math.sin(latRad) * Math.sin(latRad);
+        double sin22lat = Math.sin(2.0*latRad) * Math.sin(2.0*latRad);
+        double result = (9.780327 * (1.0 + 5.3024e-3 * sin2lat - 5.8e-6 * sin22lat) + deltaG); 
+        return new DecimalType(result);

```

Comment:
it would be really great to have a Unit-Test to proof the correctness of this calculation